app/components/goodcity/add-item-overlay.js

```diff
@@ -35,17 +37,17 @@ export default Ember.Component.extend({
     }
   ),
 
-  resolveAddItemPromises() {
-    const loadingView = getOwner(this)
-      .lookup("component:loading")
-      .append();
-    let promises = this.addItemPromises();
-    Ember.RSVP.all(promises)
-      .then(() => {
+  async resolveAddItemPromises() {
+    try {
+      await this.runTask(async () => {
+        const promises = this.addItemPromises();
+        await Ember.RSVP.all(promises);
         this.sendAction("onConfirm");
         this.set("open", false);
-      })
-      .finally(() => loadingView.destroy());
+      }, ERROR_STRATEGIES.RAISE);
+    } catch (err) {
+      throw error;
```

Comment:
no point in catching an error just to throw it again.
Plus, the error is named `err` not `error`

Reply:
just remove the try catch and let it propagate if that's the intention